interfaces/machine_parameters/ring.py

```diff
@@ -0,0 +1,598 @@
+        self.radius = self.circumference / (2 * np.pi)
+
+        # Computing the revolution period on the design orbit
+        # as well as revolution frequency and angular frequency
+        self.t_rev_design = np.dot(self.section_length_design,
+                                   1 / (self.beta * c))
+        self.f_rev_design = 1 / self.t_rev_design
+        self.omega_rev_design = 2 * np.pi * self.f_rev_design
+
+        # Computing the time of flight in each section
+        # and the revolution period on the beam orbit including
+        # possible orbit bumps
+        self.t_rev = np.sum(np.array(
+            self.section_length / (self.beta * c)), axis=0)
+        self.f_rev = 1 / self.t_rev
+        self.omega_rev = 2 * np.pi * self.f_rev
+
+        # Recalculating the delta_E
+        if self.n_turns > len(self.use_turns):
+            self.delta_E = np.zeros(self.energy.shape)
+            self._recalc_delta_E()
+        else:
+            self.delta_E = np.diff(self.energy, axis=1)
+
+        # Determining the momentum compaction orders defined in all sections
+        # The orders 1 and 2 are presently set by default to zeros if
+        # not defined in sections for the calculation of all orders of eta
+        self.alpha_orders = []
+        for section in self.RingSection_list:
+            self.alpha_orders += section.alpha_orders
+
+        # Add alpha orders if eta_orders is defined
+        self.eta_orders = kwargs.pop('eta_orders', 0)
+        if self.eta_orders >= 1:
+            self.alpha_orders += [1]
+        if self.eta_orders >= 2:
+            self.alpha_orders += [2]
+        if self.eta_orders >= 3:
+            warn_message = 'The eta_orders can only be computed up to eta_2!'
+            warnings.warn(warn_message)
+
+        self.alpha_orders = np.unique(self.alpha_orders)
+
+        # Getting the momentum compaction programs in all sections
+        # and combining the programs (the missing programs are filled
+        # with zeros).
+        # The programs are reshaped to the size of the momentum program
+        for order in self.alpha_orders:
+
+            alpha_prog = []
+            alpha_name = 'alpha_%d' % (order)
+
+            for section in self.RingSection_list:
+
+                try:
+                    alpha_prog.append(getattr(section, alpha_name))
+                except AttributeError:
+                    alpha_prog.append(ring_programs.momentum_compaction(
+                        0, order=order))
+
+            alpha_prog = ring_programs.momentum_compaction.combine_single_sections(
+                *alpha_prog, interpolation='linear')
+
+            setattr(self, alpha_name, alpha_prog.reshape(
+                self.n_sections, self.cycle_time, self.use_turns))
+
+        # Slippage factor derived from alpha, beta, gamma
+        for order in range(self.eta_orders + 1):
+            setattr(self, 'eta_%d' % (order), np.zeros(self.momentum.shape))
+        self._eta_generation()
+
```

Comment:
By this point the kwargs should be empty.  Maybe worth putting something like:
```python
if len(kwargs) > 0:
    warings.warn(f"Unused kwargs have been detected, they are {list(kwargs.keys())}")
```

Reply:
Added in both Ring and RingSection, this can be generalized everywhere I guess

Reply:
Depends how they're used.  If we have a specific set of required kwargs and we pop them all out then we can be sure it's empty by the end.  If they get passed on to other functions and objects we can't know that.  Where it's possible I think we should aim to include it though.